### ISSUE TITLE
`CLOrient3D` parameter `n_check` validation

### ISSUE THREAD

**Issue by j-c-c** [ComputationalCryoEM/ASPIRE-Python#966]
The `n_check` parameter is used in `build_clmatrix()` to reduce the number of pairs to find common lines between and in `estimate_shifts()` to reduce memory usage.

There are a couple issues with this parameter that need investigating:
1) The parameter is not currently configurable from any of the subclasses that actually implement a common-lines method.
2) It is not clear that this parameter is needed in `build_clmatrix()`. If it is, we should validate that the CL algos still work with it.
3) We should validate that there is actually a memory issue when using `estimate_shifts()`.

**Comment by garrettwrong** [comment]
@j-c-c , is this still an active issue? Thanks

**Comment by j-c-c** [comment]
@garrettwrong, yes, I will try to address when I work on validating the `estimate_shifts` method (#995).
